engineapi/instance_manager.go

```diff
@@ -195,21 +238,32 @@ func NewInstanceManagerClient(im *longhorn.InstanceManager) (*InstanceManagerCli
 	// Create a new instance service  client
 	endpoint = "tcp://" + imutil.GetURL(im.Status.IP, InstanceManagerInstanceServiceDefaultPort)
 	instanceServiceClient, err := initInstanceServiceTLSClient(endpoint)
+	defer func() {
+		if err != nil && instanceServiceClient != nil {
+			instanceServiceClient.Close()
+			instanceServiceClient = nil
+		}
+	}()
```

Comment:
We already have a cleanup inside `initInstanceServiceTLSClient`, so why we need the cleanup again here?

Reply:
The cleanup inside `initInstanceServiceTLSClient` closes `instanceServiceClient` created in the function.
The defer function here is for the non-tls client instead.